core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -584,16 +584,21 @@ public void runInContext() {
 
     @Override
     public void closed(final Status status) {
+      // For cancellations, promptly inform any users of the context that their work should be
+      // aborted. Otherwise, we can wait until pending work is done.
+      if (!status.isOk()) {
+        context.cancel(status.getCause());
+      }
+      // The contract of this.listener is that it's always accessed from inside callExecutor.
```

Comment:
Isn't this comment true for the entire class? It seems it's the entire purpose for the class. It seems like either remove it or move it to getListener()/listener or some similar class-wide documentation.